File: api/src/models/models/jabatanModel.js
module.exports = (sequelize, DataTypes) => {
  const Jabatan = sequelize.define('Jabatan', 
  {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    nama: {
      type: DataTypes.STRING,
      unique: true,
      allowNull: false
    },
    gaji_pokok: {
      type: DataTypes.DECIMAL(12,2),
      allowNull: false,
      defaultValue: 0.00
    },
    tunjangan_jabatan: {
      type: DataTypes.DECIMAL(12,2),
      allowNull: false,
      defaultValue: 0.00
    },
    status: {
      type: DataTypes.STRING,
      allowNull: false,
      defaultValue: 'ACTIVE'
    }
  }, {
    tableName: 'jabatan',
    createdAt: 'created_at',
    updatedAt: 'updated_at'
  })

  return Jabatan
}

File: flutter/lib/models/jabatan_model.dart
class Jabatan {
  int? id;
  String? nama;
  String? gajiPokok;
  String? tunjanganJabatan;
  int? createdBy;
  int? updatedBy;

  Jabatan.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nama = json['nama'];
    gajiPokok = json['gaji_pokok'];
    tunjanganJabatan = json['tunjangan_jabatan'];
    createdBy = json['created_by'];
    updatedBy = json['updated_by'];
  }
}